credential-stores/postgres/main.go

```diff
@@ -75,3 +76,22 @@ func NewPostgres(ctx context.Context) (common.Database, error) {
 
 	return common.NewDatabase(ctx, db)
 }
+
+func authenticatedHandler(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if err := authenticate(r.Header); err != nil {
+			http.Error(w, err.Error(), http.StatusUnauthorized)
+			return
+		}
+		next(w, r)
+	}
+}
+
+func authenticate(headers http.Header) error {
+	token := headers.Get("X-GPTScript-Daemon-Token")
+	if token != os.Getenv("GPTSCRIPT_DAEMON_TOKEN") {
```

Comment:
nit: can this be refactored to only read the environment variable once?

Reply:
fixed